client/spring-zeebe-starter/src/main/java/io/camunda/zeebe/spring/client/actuator/MicrometerMetricsRecorder.java

```diff
@@ -0,0 +1,65 @@
+package io.camunda.zeebe.spring.client.actuator;
+
+import io.camunda.zeebe.spring.client.connector.MetricsRecorder;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.autoconfigure.endpoint.EndpointAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.stereotype.Component;
+
+import java.lang.invoke.MethodHandles;
+import java.util.HashMap;
+import java.util.Map;
+
+public class MicrometerMetricsRecorder implements MetricsRecorder {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  private final MeterRegistry meterRegistry;
+
+  private final Map<String, Counter> counterActivated = new HashMap<>();
+  private final Map<String, Counter> counterCompleted = new HashMap<>();
+  private final Map<String, Counter> counterFailed = new HashMap<>();
+  private final Map<String, Counter> counterBpmnError = new HashMap<>();
```

Comment:
Isn't it enough to just have Counter objects here for each type? Either that or one map with a counter per action.

Reply:
Yeah  -the latter. The recorder can handle multiple metric names, currently one for JOB and one cor CONNECTOR_INVOCATIONS. I am pretty sure we will also add something for INBOUND. I agree though, that it might not be a design super easy to follow - let me adjust it